pkg/detectors/aws/aws.go

```diff
@@ -180,20 +178,36 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			if err == nil {
 				s1.ExtraData["account"] = account
 			}
-			if !s.verifyCanaries && !strings.Contains(dataStr, "Redacted: "+resIDMatch) {
-				if _, ok := thinkstCanaryList[account]; ok {
-					s1.ExtraData["is_canary"] = "true"
-					s1.ExtraData["message"] = thinkstMessage
+			if _, ok := thinkstCanaryList[account]; ok {
+				s1.ExtraData["is_canary"] = "true"
+				s1.ExtraData["message"] = thinkstMessage
+				verified, arn, err := s.verifyCanary(resIDMatch, resSecretMatch)
```

Comment:
oh @joeleonjr you raised an excellent point internally - should we be doing this if the user is running with verification disabled?